src/app/pages/AccountDetailsPage/hook.ts

```diff
@@ -0,0 +1,73 @@
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import {
+  EvmTokenType,
+  Layer,
+  RuntimeEvmBalance,
+  useGetRuntimeAccountsAddress,
+  useGetRuntimeTransactions,
+} from '../../../oasis-indexer/api'
+import { useLoaderData } from 'react-router-dom'
+import { AppErrors } from '../../../types/errors'
+import { useSearchParamsPagination } from '../../components/Table/useSearchParamsPagination'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+
+export const useAccount = () => {
+  const { network, layer } = useRequiredScopeParam()
+  if (layer === Layer.consensus) {
+    // There can be no ERC-20 or ERC-721 tokens on consensus
+    throw AppErrors.UnsupportedLayer
+  }
+  const address = useLoaderData() as string
+  const query = useGetRuntimeAccountsAddress(network, layer, address!)
+  const account = query.data?.data
+  const isLoading = query.isLoading
+  const tokenBalances: Record<EvmTokenType, RuntimeEvmBalance[]> = {
+    ERC1155: [],
+    ERC20: [],
+    ERC721: [],
+    OasisSdk: [],
+  }
+  if (!isLoading && account) {
+    account.evm_balances.forEach(balance => {
+      if (balance.token_type) tokenBalances[balance.token_type].push(balance)
+    })
+  }
```

Comment:
Can we avoid another hooks layer of API processing?
I would move this into api.ts, or make it a pure function 
`parseAccount(query) => { isLoading, account: { address, balances, tokenBalances, ... } }`

Reply:
Good idea, moving it to  api.ts

Reply:
Done.